src/activemasternode.cpp

```diff
@@ -113,6 +113,11 @@ std::string CActiveMasternode::GetStatus()
 
 bool CActiveMasternode::SendMasternodePing(std::string& errorMessage)
 {
+    if (!vin.is_initialized()) {
```

Comment:
let's better compare it against `nullopt` to know whether is initialized or not. 

The main goal of the Optional class is to move away from boost internal methods, to make smoother the transition to c++17 std::optional.

Reply:
fair enough.